glib/src/subclass/types.rs

```diff
@@ -369,6 +369,31 @@ pub trait ObjectSubclass: Sized + 'static {
     fn with_class(_klass: &Self::Class) -> Self {
         Self::new()
     }
+
+    /// Performs additional instance initialization.
+    ///
+    /// Called just after `with_class()`. At this point the initialization has not completed yet, so
+    /// only a limited set of operations is safe (see `InitializingObject`).
+    fn instance_init(_obj: InitializingObject<Self::Type>) {}
```

Comment:
```suggestion
    fn instance_init(_obj: &InitializingObject<Self::Type>) {}
```

You probably don't want it to be moved. That provides even more ways of doing wrong things :)

Reply:
Hm, what for example?

Reply:
In case of `Borrow<T>` you could carry around a dangling pointer, for example.

Reply:
Right, good point.